Fix RxSession in Testkit Backend

bigmontz opened the pull request:
Testkit environment variables without `TEST_` prefix are not forward by the Testkit to the
scripts in the driver. So we need change the session type variable expected by the `backend.py`
to  be `TEST_SESSION_TYPE` and then remove the prefix before redirect to the backend.

This PR also fix the `SessionRun` and `TransactionRun` to not initiate the stream.

**Warn**: `RxSession.beginTransaction` and `RxSession.execute*` were tweaked. The `Observable<RxTransaction>` only emits the `RxTransaction` when the begin is executed with success. The begin still defer the execution to the point observer is subscribed.

bigmontz committed:
Fix RxSession in Testkit Backend

Testkit environment variables without `TEST_` prefix are not forward by the Testkit to the
scripts in the driver. So we need change the session type variable expected by the `backend.py`
to  be `TEST_SESSION_TYPE` and then remove the prefix before redirect to the backend.

This PR also fix the `SessionRun` and `TransactionRun` to not initate the stream.

bigmontz committed:
Fix TransactionRun and SessionRun in Rx backend

Improves the RxSession.beginTransaction by make it failing in the correct observable.

bigmontz commented:
Depends on: https://github.com/neo4j-drivers/testkit/pull/512

fbiville commented on packages/neo4j-driver/src/result-rx.js:
```diff
@@ -35,8 +35,9 @@ export default class RxResult {
    * @constructor
    * @protected
    * @param {Observable<Result>} result - An observable of single Result instance to relay requests.
+   * @param {number} state - The streamming state
```
```suggestion
   * @param {number} state - The streaming state
```

fbiville commented on packages/testkit-backend/src/feature/common.js:
```diff
@@ -33,7 +33,7 @@ const features = [
   'Feature:API:ConnectionAcquisitionTimeout',
   'Feature:API:Driver:GetServerInfo',
   'Feature:API:Driver.VerifyConnectivity',
-  'Feature:API:Result.Peek',
```
why was this one removed?

bigmontz replied:
Because reactive doesn't support Peek. This feature is enable for async at `packages/testkit-backend/src/feature/async.js`

fbiville commented on packages/testkit-backend/src/skipped-tests/rx.js:
```diff
@@ -4,6 +4,15 @@ const skippedTests = [
     ifEquals('stub.disconnects.test_disconnects.TestDisconnects.test_disconnect_on_tx_begin')
+  ),
+  skip(
+    'Backend could not support this test',
```
what happened with these?

bigmontz replied:
The error occurs in tricky place in the reactive backend. The reactive backend implements a custom subscriber to convert the reactive stream to `AsyncIterator`, the error happens there and it happens in a way it is difficult to align with the testkit premises.

robsdedude replied:
But is there a reasonable way for users to handle this? Because there sure should be a way how users can handle unknown tz ids coming from the server.

bigmontz replied:
It should have, I will double check it again.

bigmontz replied:
It was actually the driver log causing issues. When the debug is turned on, the driver tries to access the broken object and it moves the point where the error occurs (it happens earlier, before the user get access to the record). 

I made the broken object a bit more lenient and I added a especial serialisation for this kind of object in the `json.stringify` wrapper.

bigmontz committed:
Apply suggestions from code review

Co-authored-by: Florent Biville <445792+fbiville@users.noreply.github.com>

bigmontz committed:
Fix serialization of broken objects in the driver logs

Trying to logging broken objects was causing the driver fails in unexpected moments.
This broken behaviour makes the driver fail different when the `debug` log is enabled.
